### ISSUE TITLE
textAlign(RIGHT) including spaces at the end of each line

### ISSUE THREAD

**Issue by lilalinda** [processing/processing#3028]
When using the example code below -- inspired by the `text()` API entry -- the rendering shows a slight offset from the right.  Without digging deeper into the code, I would assume these to be the white spaces where the lines are allowed to wrap.  When setting alignment to `RIGHT`, in my opinion, the spaces should move to the following line or swallowed entirely to allow for perfect alignment.

```
void setup() {
  size(100,100);
  smooth();
  noLoop();
}

void draw() {
  String s = "The quick brown fox jumped over the lazy dog. X";
  fill(0);
  textAlign(RIGHT);
  text(s, 10, 10, 70, 80);  // Text wraps within text box
}
```
The orange line shows the unwanted space, which should align better with the last character X:

![sketch_150103a](https://cloud.githubusercontent.com/assets/1231235/5603725/4dffc02e-9359-11e4-899a-730461342124.png)

**Comment by spasmsaps** [comment]
IMHO, it would be nice to have control in how text is wrapped. textWrapMode(int MODE)
http://www.w3.org/TR/css3-text/#white-space-property

**Comment by spasmsaps** [comment]
related #478

**Comment by benfry** [comment]
@lilalinda Huh, I can't imagine how that's never been seen/caught before.

**Comment by Akarshit** [comment]
I was looking into this issue but i am facing some problems , i asked it on the forum but haven't received any response yet . I am really stuck and if someone would be kind enough to take a look at   http://forum.processing.org/two/discussion/9245/how-to-make-changes-to-papplet-class-in-processing-core-papplet-java#latest  
I would be really grateful.
I know this might not be the right place to reference the forum discussion , but i am facing similar problems in changing other files .

**Comment by benfry** [comment]
@Akarshit added notes to that forum post, thanks for looking into it.

**Comment by Akarshit** [comment]
@benfry I tested few texts with textAlign(LEFT) and they were working fine.

Could you paste here a string which gives incorrect output.

**Comment by benfry** [comment]
Fixed since 3.0a6.
